src/libAtomVM/bitstring.h

```diff
@@ -379,6 +419,23 @@ static inline bool bitstring_insert_utf8(term dst_bin, size_t offset, avm_int_t
     return bitstring_utf8_encode(c, dst, out_size);
 }
 
+/**
+ * @brief Match a character in UTF-8 format
+ *
+ * @param src_bin binary to match against
+ * @param offset offset, in bits, to where to start to match the character
+ * @param c int to decode to
+ * @param out_size the size in bytes, on output
+ * @return \c true if encoding was successful, \c false if src_bin at offset is not a valid
+ * unicode character
+ */
+static inline bool bitstring_match_utf8(term src_bin, size_t offset, int32_t *c, size_t *out_size)
```

Comment:
same here, let's use `const` keyword if we don't plan to write to `c` buffer. Same comment applies also to any other similar function.

Reply:
This function does modify *c.